lms/views/dashboard/api/grading.py

```diff
@@ -86,7 +87,17 @@ def create_grading_sync(self):
     def get_grading_sync(self):
         assignment = self.dashboard_service.get_request_assignment(self.request)
         if grading_sync := self.auto_grading_service.get_last_sync(assignment):
-            return {"status": grading_sync.status}
+            last_successful_sync = (
+                grading_sync
+                if grading_sync.status == AutoGradingSyncStatus.FINISHED
+                else self.auto_grading_service.get_last_successful_sync(assignment)
+            )
+            return {
+                "status": grading_sync.status,
+                "last_successful_sync_date": last_successful_sync.created.isoformat()
+                if last_successful_sync != None
+                else None,
+            }
```

Comment:
I tried to add a return type annotation in this method, but mypy started to complain due to the two different "shapes" this method can return, with the `message` prop for errors, or the rest for successful checks.

I tried to define the return type as a union with `APIGradingSync | APIGradingSyncError` and  `Union[APIGradingSync, APIGradingSyncError]`, but none worked.

I ended up just adding the new prop, and we can improve typing afterwards, as I'm probably missing something.